docs/kyma/docs/037-gs-installation-overrides.md

```diff
@@ -0,0 +1,168 @@
+
+
+### Common vs component overides
+
+Installer looks for available overrides each time a component installation/update operation is due.
+Overrides for the component are composed from two sets: **common** overrides and **component-specific** overrides.
+- Common overrides are used for installation of all components. They are defined in ConfigMaps/Secrets marked with `installer:overrides` label only.
+- Component-specific overrides are used only for the installation of the specific component. They are defined in ConfigMaps/Secrets marked with both: `installer:overrides` and `component: <name>` labels, where `<name>` is the component name. Component-Specific overrides have precedence over Common ones in case of conflicting entries.
+
+
+### Examples
+
+
+#### Top-level charts overrides
+
+Overrides for top-level charts are straightforward. Just use template value from the chart (without leading ".Values." prefix) as the entry key in ConfigMap/Secret.
+
+Example:
+
+There's a `core` top-level chart, that is installed by the Installer.
+In one of it's templates there's a following line with a value reference:
+```
+memory: {{ .Values.test.acceptance.ui.requests.memory }}
+```
+In order to resolve this template, there's a default value for "test.acceptance.ui.requests.memory" in the `values.yaml` file of the chart.
+The following fragment of `values.yaml` shows this definition:
+```
+test:
+  acceptance:
+    ui:
+      requests:
+        memory: "1Gi"
+```
+
+If you want to override this value, for example to "2Gi", proceed as follows:
+- Create a ConfigMap in `kyma-installer` namespace, labelled with: `installer:overrides` (or reuse an existing one)
+- Add an entry `test.acceptance.ui.requests.memory: 2Gi` to the map.
+
+Once the installation starts, Installer generates overrides based on the map entries and value of "2Gi" will be used instead of default "1Gi" from the chart `values.yaml` file.
+For values that should be kept in Secrets, just define a Secret object instead of ConfigMap with the same key and value. Don't forget to label the Secret with `installer:overrides` and remember that values in Secrets must be base64-encoded.
+
+
+#### Sub-chart overrides
+
+Overrides for sub charts follow the same convention as top-level charts, but require additional information about sub-chart location.
+When a sub-chart contains `values.yaml` file, the information about chart location is not necessary because chart and it's `values.yaml` file are on the same level in directory hierarchy.
+The situation is different when the Installer installs a chart with sub-charts.
+All template values for a sub-chart must be prefixed with a sub-chart "path" (relative to top-level "parent" chart).
+This in not Installer-specific requirement, the same considerations apply when you provide overrides manually via `helm` command-line tool.
+
+Example:
+
+There's a `core` top-level chart, that is installed by the Installer.
```

Comment:
There's a `core` top-level chart, that the Installer installs.
There's an `application-connector` sub-chart in `core` with another nested sub-chart: `connector-service`.
In one of its templates there's a following fragment (shortened):